datadog/data_source_datadog_cloud_workload_security_agent_rules.go

```diff
@@ -0,0 +1,76 @@
+package datadog
+
+import (
+	"context"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceDatadogCloudWorkloadSecurityAgentRules() *schema.Resource {
+	return &schema.Resource{
+		Description: "Use this data source to retrieve information about existing cloud workload security agent rules for use in other resources.",
+		ReadContext: dataSourceDatadogCloudWorkloadSecurityAgentRulesRead,
+
+		Schema: map[string]*schema.Schema{
+			// Computed
+			"agent_rules_ids": {
+				Description: "List of IDs of agent rules.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"agent_rules": {
+				Description: "List of agent rules.",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: cloudWorkloadSecurityAgentRuleSchema(),
```

Comment:
Instead of reusing the schema, I would redefine it in the data source and add the id's within it. Additionally, lets set all of the attributes to computed as well. For example: https://github.com/DataDog/terraform-provider-datadog/blob/master/datadog/data_source_datadog_monitors.go#L44

Reply:
Done